### DISCUSSION TITLE
Support Bullet Points

### DISCUSSION THREAD

**Discussion by michaelPotter** [cooklang/spec#60] [Ideas]
Hi, I'd like to structure recipe ingredients or sub-steps as bullet points where possible. Right now when I try to do bullet points, the cli and server recipe views separate each bullet point into it's own step.

I don't have the best working memory, so things separated into bullets on their own lines makes it easier for me to keep track of which bits I've already done and where I'm at in the recipe. I tend to get lost in the paragraphs when it's formed as prose.

Guacamole is a bit of a contrived example, but I personally find this much easier to follow (and not miss an ingredient lol):
```
In a medium bowl, mash together:
- @avacados{3} - peeled, pitted, and mashed
- juice of @lime{1}
- @salt

Mix in:
- @onion{1/2%cup} diced
- @fresh cilantro{3%tablespoons} chopped
- @roma tomatoes{2}
- @garlic{1%tsp} minced
- @ground cayenne pepper{1%pinch}

Refrigerate 1 hour for best flavor, or serve immediately
```
than this:
```
In a medium bowl, mash together @avacados{3} (peeled, pitted, and mashed), juice of one @lime{1}, @salt

Mix in diced @onion{1/2%cup} diced, chopped @fresh cilantro{3%tablespoons}, @roma tomatoes{2}, minced @garlic{1%tsp}, @ground cayenne pepper{1%pinch}

Refrigerate 1 hour for best flavor, or serve immediately
```

**Comment by dubadub** [comment]
Hi @michaelPotter thanks for bringing this up.

I think there're two problems to solve here.

1. In text editor the language plugin should set word wrap by default. For example when writing in Markdown it does it for you and text displayed on multiple lines.
2. Need for syntax to continue step on next line. Example borrowed from bash would be:
```
In a medium bowl, mash together: \
- @avacados{3} - peeled, pitted, and mashed \
- juice of @lime{1} \
- @salt
```
In that case we keep freedom of expression for a user and they can use "bullet" points style or any other. 

What do you think?

**Comment by michaelPotter** [comment]
Hi, I like the backslash syntax. That would keep things flexible so that you can use whatever type of bullet points you want. To me, that seems better than having a number of predefined bullet options the way markdown does.

I'm not sure I understand your point 1 though... Are you suggesting that multiple consecutive lines should be wrapped into one step entry; and separate steps should be separated by an empty line - the way markdown does it? I think that would be good - seems like that would be less surprising to someone familiar with markdown.

Or do you just mean that the text editor should soft wrap so that there aren't run on lines like that?

**Comment by dubadub** [comment]
Apparently in Markdown (it’s so intuitive that you do it without much thinking) they only split paragraphs if there’s an empty line between blocks of text ([source](https://daringfireball.net/projects/markdown/syntax#p)). 

I think it makes sense to use a similar approach in Cooklang too and split into steps only if there’s an empty line. 

Re: https://github.com/cooklang/spec/issues/45.

**Comment by dubadub** [comment]
This is now part of the main specification, closing the discussion.
